spammer.go

```diff
@@ -0,0 +1,160 @@
+package main
+
+import (
+	"fmt"
+	"sort"
+	"sync"
+)
+
+func RunPipeline(cmds ...cmd) {
+	var out, in chan interface{}
+	in = make(chan interface{})
+	wg := &sync.WaitGroup{}
+
+	for _, c := range cmds {
+		out = make(chan interface{})
+		wg.Add(1)
+		go func(c cmd, in, out chan interface{}) {
+			defer wg.Done()
+			c(in, out)
+			close(out)
+		}(c, in, out)
+		in = out
+	}
+	wg.Wait()
+}
+
+func SelectUsers(in, out chan interface{}) {
+	var seen sync.Map
+	var wg sync.WaitGroup
+
+	for email := range in {
+		wg.Add(1)
+		go func(email interface{}) {
+			defer wg.Done()
+
+			emailStr := email.(string)
+
+			alias := usersAliases[emailStr]
+
+			if alias != "" {
+				emailStr = alias
+			}
```

Comment:
Обычно, в гощке используют проверку наличия в map

```suggestion
			alias, found := usersAliases[emailStr]

			if found {
				emailStr = alias
			}
```